src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -29,22 +29,15 @@ public virtual async Task<ExecutionResult> ExecuteAsync(ExecutionContext context
             // After the entire node tree has been executed, get the values
             object? data = rootNode.PropagateNull() ? null : rootNode;
 
-            return new ExecutionResult
+            return new ExecutionResult(context)
             {
                 Executed = true,
                 Data = data,
-                Query = context.Document.Source,
-                Document = context.Document,
-                Operation = context.Operation,
-                Extensions = context.OutputExtensions
             };
         }
 
-        /// <summary>
-        /// Executes an execution node and all of its child nodes. This is typically only executed upon
-        /// the root execution node.
-        /// </summary>
-        protected abstract Task ExecuteNodeTreeAsync(ExecutionContext context, ObjectExecutionNode rootNode);
+        /// <inheritdoc/>
+        public abstract Task ExecuteNodeTreeAsync(ExecutionContext context, ExecutionNode rootNode);
```

Comment:
Does really `ExecutionNode rootNode` parameter still _root_ ?

Reply:
Depends on the point of view.  As it relates to the "tree" (this is a method called "Execute Node Tree") I would say it is the root of the tree.

As it relates to a query or mutation, yes it will always be the root AST node.  As it relates to a subscription request, it's not technically the root of the entire query AST, but it is the "root field" as described in the spec here: http://spec.graphql.org/October2021/#sec-Single-root-field

If you would like to rename it, that's fine with me, but it seems best to me if we leave it as-is.

Reply:
OK, leave as-is.